tcl/org.testeditor.tcl.dsl.tests/src/org/testeditor/tcl/dsl/jvmmodel/CallTreeBuilderTest.xtend

```diff
@@ -1,166 +1,291 @@
 			steps += specificationStep('My', 'first', 'test', 'step')
 			cleanup += createTestCleanup
 		]
-		
+
 		// when
 		val actualTree = builderUnderTest.buildCallTree(testCase)
-		
-		//then
+
+		// then
 		actualTree.children => [
 			assertSize(3)
 			get(0).displayname.assertEquals('Setup')
 			get(1).displayname.assertEquals('My first test step')
 			get(2).displayname.assertEquals('Cleanup')
 		]
 	}
-	
+
 	@Test
 	def void returnsProperCallTreeForTestCaseWithComponentContext() {
-		//given
+		// given
 		val testCase = testCase('TheTestCase') => [
 			steps += specificationStep('My', 'first', 'test', 'step') => [
-				contexts += componentTestStepContext(amlModel.myDialog)
+				contexts += componentTestStepContext(amlComponentForTesting)
 			]
 		]
-		
+
 		// when
 		val actualTree = builderUnderTest.buildCallTree(testCase)
-		
-		//then
+
+		// then
 		actualTree.children.head.children => [
 			assertSize(1)
-			head.displayname.assertEquals(amlModel.myDialog.name)
+			head.displayname.assertEquals(amlComponentForTesting.name)
 		]
 	}
-	
+
 	@Test
 	def void returnsProperCallTreeForTestCaseWithMacroContext() {
-		//given		
+		// given		
 		val testCase = testCase('TheTestCase') => [
 			steps += specificationStep('My', 'first', 'test', 'step') => [
 				contexts += macroTestStepContext(macroCollection('MyMacroCollection'))
 			]
 		]
-		
+
 		// when
 		val actualTree = builderUnderTest.buildCallTree(testCase)
-		
-		//then
+
+		// then
 		actualTree.children.head.children => [
 			assertSize(1)
 			head.displayname.assertEquals('MyMacroCollection')
 		]
 	}
-	
+
 	@Test
 	def void returnsProperCallTreeForTestCaseWithTestStepsInComponentContext() {
-		//given
+		// given
 		val testCase = testCase('TheTestCase') => [
 			steps += specificationStep('My', 'first', 'test', 'step') => [
-				contexts += componentTestStepContext(amlModel.myDialog) => [
+				contexts += componentTestStepContext(amlComponentForTesting) => [
 					steps += testStep('Wait', 'for').withParameter('60').withText('seconds')
 				]
 			]
 		]
-		
+
 		// when
 		val actualTree = builderUnderTest.buildCallTree(testCase)
-		
-		//then
+
+		// then
 		actualTree.children.head.children.head.children => [
 			assertSize(1)
 			head.displayname.assertEquals('Wait for "60" seconds')
 		]
 	}
-	
-}
\ No newline at end of file
+
+	@Test
+	def void returnsProperCallTreeForTestCaseWithTestStepWithAssignment() {
+		// given
+		val testCase = testCase('TheTestCase') => [
+			steps += specificationStep('My', 'first', 'test', 'step') => [
+				contexts += componentTestStepContext(amlComponentForTesting) => [
+					steps += testStepWithAssignment('myVar', 'Read', 'jsonObject', 'from').withElement('bar')
+				]
+			]
+		]
+
+		// when
+		val actualTree = builderUnderTest.buildCallTree(testCase)
+
+		// then
+		actualTree.children.head.children.head.children => [
+			assertSize(1)
+			head.displayname.assertEquals('com.google.gson.JsonObject myVar = Read jsonObject from <bar>')
```

Comment:
Type of variable in parenthesis at the end? It is more of an implementation detail and should not be presented to the Tester, right?

Reply:
… it's now `myVar = Read jsonObject from <bar> [com.google.gson.JsonObject]`; this format is also used for `reporterEnterCall(…)` (see above), and there were several test cases covering this, which I adapted, accordingly.

I suppose another option to consider would be to only put the simple type into the log message (i.e. substring starting from lastIndexOf('.'), leaving only `[JsonObject]` as type info suffix), as the type info can get quite long, e.g. for types with type parameters (generics).